pubsub/iterator.go

```diff
@@ -438,6 +439,14 @@ func (it *messageIterator) sendModAck(m map[string]bool, deadline time.Duration)
 				recordStat(it.ctx, ModAckTimeoutCount, 1)
 				return nil
 			default:
+				if err == nil {
+					return nil
+				}
+				// This addresses an error where `context deadline exceeded` errors
+				// not captured by the previous case do not cause fatal errors.
+				if strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
Yeah, so I have reason to believe that the `context deadline exceeded` error might be coming from something in the library rather than from grpc-go, so there isn't an associated code here. In this case, the fake simulates the error with an `Unknown context deadline exceeded`, so it doesn't 100% mimic the error users are seeing.